cirq/google/optimizers/optimize_for_gmon.py

```diff
@@ -0,0 +1,109 @@
+# limitations under the License.
+"""A combination of several optimizations targeting XmonDevice."""
+from typing import Callable, cast, List, Optional, TYPE_CHECKING
+
+from cirq import circuits, devices, optimizers
+from cirq.google.optimizers import (convert_to_xmon_gates,
+                                    ConvertToSycamoreGates,
+                                    ConvertToSqrtIswapGates)
+
+if TYPE_CHECKING:
+    import cirq
+
+_TOLERANCE = 1e-5
+
+
+def _merge_rots(c: 'cirq.Circuit'):
+    return optimizers.merge_single_qubit_gates_into_phased_x_z(c, _TOLERANCE)
+
+
+_XMON_OPTIMIZERS: List[Callable[['cirq.Circuit'], None]] = [
+    convert_to_xmon_gates.ConvertToXmonGates().optimize_circuit,
+    optimizers.MergeInteractions(tolerance=_TOLERANCE,
+                                 allow_partial_czs=False).optimize_circuit,
+    _merge_rots,
+    optimizers.EjectPhasedPaulis(tolerance=_TOLERANCE).optimize_circuit,
+    optimizers.EjectZ(tolerance=_TOLERANCE).optimize_circuit,
+    optimizers.DropNegligible(tolerance=_TOLERANCE).optimize_circuit,
+]
+
+_XMON_OPTIMIZERS_PART_CZ: List[Callable[['cirq.Circuit'], None]] = [
+    convert_to_xmon_gates.ConvertToXmonGates().optimize_circuit,
+    optimizers.MergeInteractions(tolerance=_TOLERANCE,
+                                 allow_partial_czs=True).optimize_circuit,
+    _merge_rots,
+    optimizers.EjectPhasedPaulis(tolerance=_TOLERANCE).optimize_circuit,
+    optimizers.EjectZ(tolerance=_TOLERANCE).optimize_circuit,
+    optimizers.DropNegligible(tolerance=_TOLERANCE).optimize_circuit,
+]
+
+_SYCAMORE_OPTIMIZERS = [
+    ConvertToSycamoreGates().optimize_circuit,
+    _merge_rots,
+    optimizers.EjectPhasedPaulis(tolerance=_TOLERANCE).optimize_circuit,
+    optimizers.EjectZ(tolerance=_TOLERANCE).optimize_circuit,
+    optimizers.DropNegligible(tolerance=_TOLERANCE).optimize_circuit,
+]  # type: List[Callable[[circuits.Circuit], None]]
+
+_SQRT_ISWAP_OPTIMIZERS = [
+    ConvertToSqrtIswapGates().optimize_circuit,
+    _merge_rots,
+    optimizers.EjectPhasedPaulis(tolerance=_TOLERANCE).optimize_circuit,
+    optimizers.EjectZ(tolerance=_TOLERANCE).optimize_circuit,
+    optimizers.DropNegligible(tolerance=_TOLERANCE).optimize_circuit,
+]  # type: List[Callable[[circuits.Circuit], None]]
+
+_OPTIMIZER_TYPES = {
+    'xmon': _XMON_OPTIMIZERS,
+    'xmon_partial_cz': _XMON_OPTIMIZERS_PART_CZ,
+    'sqrt_iswap': _SQRT_ISWAP_OPTIMIZERS,
+    'sycamore': _SYCAMORE_OPTIMIZERS,
+}
+
+
+def optimized_for_gmon(circuit: 'cirq.Circuit',
+                       new_device: Optional['cirq.google.XmonDevice'] = None,
+                       qubit_map: Callable[['cirq.Qid'], devices.GridQubit] =
+                       lambda e: cast(devices.GridQubit, e),
+                       optimizer_type: str = 'sqrt_iswap') -> 'cirq.Circuit':
+    """Optimizes a circuit with XmonDevice in mind.
+
+  Starts by converting the circuit's operations to the xmon gate set, then
```

Comment:
It seems like xmon and gmon are being conflated here.

Reply:
Updated pydocs.